Framework/CurveFitting/src/FuncMinimizers/DTRSMinimizer.cpp

```diff
@@ -343,14 +343,12 @@ void roots_quadratic(double a0, double a1, double a2, double tol, int &nroots,
     auto pprime = two * a2 * root1 + a1;
     if (pprime != zero) {
       root1 = root1 - p / pprime;
-      p = (a2 * root1 + a1) * root1 + a0;
```

Comment:
I think the original code contained loops and `p` was updated on each iteration after the `root` was updated. They removed the loop, kept a single iteration and didn't check that `p` isn't used after that.